Assets/Shopify/UIToolkit/Theme Controllers/CartController.cs

```diff
@@ -0,0 +1,122 @@
+namespace Shopify.UIToolkit {
+    using UnityEngine.Events;
+    using Shopify.Unity;
+    using Shopify.Unity.SDK;
+    using System.Linq;
+
+    public class CartController {
+        [System.Serializable]
+        public class QuantityChangeEvent : UnityEvent<int> {}
+
+        [System.Serializable]
+        public class PurchaseFailedEvent : UnityEvent<ShopifyError> {}
+
+        private Cart Cart;
+        public UnityEvent OnPurchaseStarted = new UnityEvent();
+        public QuantityChangeEvent OnQuantityChange = new QuantityChangeEvent();
+        public UnityEvent OnPurchaseComplete = new UnityEvent();
+        public UnityEvent OnPurhcaseCancelled = new UnityEvent();
+        public PurchaseFailedEvent OnPurhcaseFailed = new PurchaseFailedEvent();
```

Comment:
We will need to expose these in an editor in the future, but its probably best to do so when doing the rest of the events